props/saber_BC_buttons.h

```diff
@@ -386,31 +1136,495 @@ EFFECT(volmax);     // for maximum volume reached
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  int current_blade_ = 1; // Start with the first blade
+  ShowColorSingleBladeTemplate<Pulsing<ShowColorStyle,Black,800>,Pulsing<ShowColorStyle,Black,800>> highlighted_blade_;
+
+  uint16_t size() override {
+    return NUM_BLADES;
+  }
+  void mode_activate(bool onreturn) override {
+    mode::getSL<SPEC>()->SayEditBladeLength();
+    mode::getSL<SPEC>()->SaySelectBlade();
+    ShowColorStyle::SetColor(Color16(65535, 65535, 65535));
+    SPEC::SteppedMode::mode_activate(onreturn);
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(current_blade_);
+    current_blade_ = (current_blade_ % NUM_BLADES) + 1;
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade_);
+    current_blade_ = (current_blade_ + NUM_BLADES - 2) % NUM_BLADES + 1;
+    highlighted_blade_.Start(current_blade_);
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade_ << "\n";
+    say();
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade_);
+  }
+
+  void select() override {
+    // Set the current blade for the ChangeBladeLengthMode before transitioning
+    // Deactivate current mode and push to LED length adjustment mode for the selected blade
+    selected_blade_ = current_blade_;
+    highlighted_blade_.Stop(current_blade_);
+    SPEC::SteppedMode::mode_deactivate();
+    pushMode<typename SPEC::ChangeBladeLengthMode>();
+  }
+
+  void exit() override {
+    highlighted_blade_.Stop(current_blade_);
+    SPEC::MenuBase::exit();
+  }
+};
+#endif  // if 0
+template<class SPEC>
+struct BCSelectBladeMode : public SPEC::MenuBase {
+  ShowColorSingleBladeTemplate<Pulsing<ShowColorStyle,Black,800>,Pulsing<ShowColorStyle,Black,800>> highlighted_blade_;
+
+  uint16_t size() override {
+    return NUM_BLADES;
+  }
+  void mode_activate(bool onreturn) override {
+    mode::getSL<SPEC>()->SayEditBladeLength();
+    mode::getSL<SPEC>()->SaySelectBlade();
+    ShowColorStyle::SetColor(Color16(65535, 65535, 65535));
+    SPEC::SteppedMode::mode_activate(onreturn);
+    highlighted_blade_.Start(this->pos_ + 1);
+    PVLOG_NORMAL << "** Highlighting blade: " << this->pos_ + 1 << "\n";
+    say();
+  }
+
+  void mode_deactivate() {
+    highlighted_blade_.Stop(this->pos_ + 1);
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(this->pos_ + 1);
+    if (++this->pos_ >= size()) this->pos_ = 0;
+    highlighted_blade_.Start(this->pos_ + 1);
+    PVLOG_NORMAL << "** Highlighting blade: " << this->pos_ + 1 << "\n";
+    say();
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(this->pos_ + 1);
+    if (this->pos_ == 0) {
```

Comment:
1482..1486 can just be SPEC::MenuBase::prev() I think.

Reply:
oh we can do that?! Call an overridden method from within the one you're overriding it with ?
Wow well that certainly makes life easier. Great. How does it look now then? (tested working).
Hopefully that wraps up this BCSelectBlade thing. 
Lots of current conversations, don't want to get lost. I'm going to stick with one at a time until good.